src/scene/pick.js

```diff
@@ -73,8 +76,8 @@ pc.extend(pc, function () {
         device.setRenderTarget(this._pickBufferTarget);
         device.updateBegin();
 
-        var pixels = new Uint8Array(4 * rect.width * rect.height);
-        device.readPixels(rect.x, rect.y, rect.width, rect.height, pixels);
+        var pixels = new Uint8Array(4 * width * height);
+        device.readPixels(x, y + height, width, height, pixels);

```

Comment:
After some testing, this actually is wrong. Render target data is still upside down, so you need to flip coordinates. Fix is simple, change this line to:
```javascript
device.readPixels(x, this._pickBufferTarget.height - (y + height), width, height, pixels);
```